packages/cheqd-sdk/src/ledger/CheqdSdkLedgerService.ts

```diff
@@ -0,0 +1,107 @@
+import type { AbstractCheqdSDKModule, CheqdSDK, ICheqdSDKOptions, DidStdFee, DIDDocument } from '@cheqd/sdk'
+import type { SignInfo } from '@cheqd/ts-proto/cheqd/did/v2'
+import type { MsgCreateResourcePayload } from '@cheqd/ts-proto/cheqd/resource/v2'
+
+import { injectable } from '@aries-framework/core'
+import { createCheqdSDK, DIDModule, ResourceModule, CheqdNetwork } from '@cheqd/sdk'
+import { DirectSecp256k1HdWallet, DirectSecp256k1Wallet } from '@cosmjs/proto-signing'
+import { EnglishMnemonic as _ } from '@cosmjs/crypto'
+import { fromString } from 'uint8arrays'
+
+import { CheqdSdkModuleConfig } from '../CheqdSdkModuleConfig'
+
+export interface CheqdSdkLedgerConfig {
+  networkType?: CheqdNetwork
+  rpcUrl?: string
+  cosmosPayerSeed?: string
+}
+
+export enum DefaultRPCUrl {
+  Mainnet = 'https://rpc.cheqd.net',
+  Testnet = 'https://rpc.cheqd.network',
+}
+
+export class EnglishMnemonic extends _ {
+  public static readonly _mnemonicMatcher = /^[a-z]+( [a-z]+)*$/
+}
+
+@injectable()
+export class CheqdSdkLedgerService {
+  private readonly cosmosPayerWallet: Promise<DirectSecp256k1HdWallet | DirectSecp256k1Wallet>
+  private sdk?: CheqdSDK
+  private fee?: DidStdFee
+  private rpcUrl: string | undefined
+
+  public constructor(config: CheqdSdkModuleConfig) {
+    const { cosmosPayerSeed, rpcUrl } = config
+    if (!cosmosPayerSeed || cosmosPayerSeed === '') {
+      this.cosmosPayerWallet = DirectSecp256k1HdWallet.generate()
+      return
+    }
+    this.cosmosPayerWallet = EnglishMnemonic._mnemonicMatcher.test(cosmosPayerSeed)
+      ? DirectSecp256k1HdWallet.fromMnemonic(cosmosPayerSeed, { prefix: 'cheqd' })
+      : DirectSecp256k1Wallet.fromKey(fromString(cosmosPayerSeed.replace(/^0x/, ''), 'hex'), 'cheqd')
+
+    this.rpcUrl = rpcUrl
```

Comment:
How feasible would it be to integrate this with the wallet from AFJ? Generally we try to avoid doing crypto not using the wallet as it makes it harder to integrate with e.g. a KMS.

Askar supports secp256k keys I believe, but otherwise we can also provide a `SigningProvider`. This way people could use more performant hardware based implemetnations when they want to.

We could then just send the bytes that should be signed to the wallet if possible?

Reply:
Will be taking this up in the next release